src/optimizer/limit_push_down_optimizer.cpp

```diff
@@ -18,15 +19,18 @@ void LimitPushDownOptimizer::visitOperator(planner::LogicalOperator* op) {
     switch (op->getOperatorType()) {
     case LogicalOperatorType::LIMIT: {
         auto& limit = op->constCast<LogicalLimit>();
-        skipNumber = limit.getSkipNum();
-        limitNumber = limit.getLimitNum();
+        if (limit.hasSkipNum()) {
```

Comment:
Was this another bug?

Reply:
Not really. It doesn't have an effect on DISTINCT. But does have an effect on the way I implement GDS. So I wouldn't call this as a bug.